app/services/samples/metadata/update_service.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    # Service used to Update Samples::Metadata
+    class UpdateService < BaseService
+      SampleMetadataUpdateError = Class.new(StandardError)
+      attr_accessor :sample, :metadata, :analysis_id
+
+      def initialize(project, sample, user = nil, params = {})
+        super(user, params)
+        @project = project
+        @sample = sample
+        @metadata = params['metadata']
+        @analysis_id = params['analysis_id']
+      end
+
+      def execute # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
+        authorize! sample.project, to: :update_sample?
+
+        if @project.id != @sample.project.id
+          raise SampleMetadataUpdateError,
+                I18n.t('services.samples.metadata.sample_does_not_belong_to_project', sample_name: @sample.name,
+                                                                                      project_name: @project.name)
+        end
+
+        if @metadata
+          # Without transforming keys, issues with overwritting can occur and multiples of the same key can appear
+          @metadata = @metadata.transform_keys(&:to_s)
+          @metadata.each do |key, value|
+            if value.blank?
+              if @sample['metadata'].key?(key)
+                @sample['metadata'].delete(key) && @sample['metadata_provenance'].delete(key)
+              end
+            else
+              update_metadata(key, value, current_user)
```

Comment:
We shouldn't need to pass the `current_user` as an argument since it's already set as an attribute accessor in the base service. You should be able to just use `current_user` in the `update_metadata` method

Reply:
Fixed in 3b73978b5eaf0aa163e17bb5034f7b9b62c8bf9b